bus-mapping/src/circuit_input_builder.rs

```diff
@@ -101,11 +101,73 @@ pub enum ExecError {
     MaxCodeSizeExceeded,
 }
 
+/// Execution state
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ExecState {
+    /// EVM Opcode ID
+    Op(OpcodeId),
+    /// Virtual step Begin Tx
+    BeginTx,
+    /// Virtual step End Tx
+    EndTx,
+    /// Virtual step Copy To Memory
+    CopyToMemory,
+}
+
+impl ExecState {
+    /// Returns `true` if `ExecState` is an opcode and the opcode is a `PUSHn`.
+    pub fn is_push(&self) -> bool {
+        if let ExecState::Op(op) = self {
+            op.is_push()
+        } else {
+            false
+        }
+    }
+
+    /// Returns `true` if `ExecState` is an opcode and the opcode is a `DUPn`.
+    pub fn is_dup(&self) -> bool {
+        if let ExecState::Op(op) = self {
+            op.is_dup()
+        } else {
+            false
+        }
+    }
+
+    /// Returns `true` if `ExecState` is an opcode and the opcode is a `SWAPn`.
+    pub fn is_swap(&self) -> bool {
+        if let ExecState::Op(op) = self {
+            op.is_swap()
+        } else {
+            false
+        }
+    }
+}
+
+/// Auxiliary data of Execution step
+#[derive(Clone, Debug)]
+pub enum StepAuxiliaryData {
+    /// Auxiliary data of Copy To Memory
+    CopyToMemory {
+        /// Source start address
+        src_addr: u64,
+        /// Destination address
+        dst_addr: u64,
+        /// Bytes left
+        bytes_left: u64,
+        /// Source end address
+        src_addr_end: u64,
+        /// If from transaction
+        from_tx: bool,
+        /// Selectors
+        selectors: Vec<u8>,
```

Comment:
found out that we don't need to provide `selectors` in the witness, and in fact, it could be derived in the witness assignment. So removed from the struct